### PR TITLE
#107 Added valueAsDate getter support

### PR DESCRIPTION
**Pull request by abaranovskii**

### CODE REVIEW

**Comment by abaranovskii** [comment]
@chemerisuk could you please take a look at issue with CI ?

**Comment by chemerisuk** [comment]
@abaranovskii I like the idea to polyfill this property, but your PR should be improved. Later I'll provide my comment inline.

File: src/main.js [hunk 1]
```diff
@@ -209,6 +209,14 @@ table+table[aria-hidden=true] {
                 set: this._setValue.bind(this, valueDescriptor.set, updateValue)
             });
 
+            var valueAsDateDescriptor = Object.getOwnPropertyDescriptor(HTMLInputElement.prototype, "valueAsDate");
```
Comment (chemerisuk) [inline src/main.js]: On IE10-11 `Object.getOwnPropertyDescriptor(HTMLInputElement.prototype, "valueAsDate")` returns a null value, so `valueAsDateDescriptor.set` will fire an error. To fix that you should not use native browser property descriptor for `valueAsDate` but always provide implementation for getter and setter instead.

**Comment by abaranovskii** [comment]
@chemerisuk done

File: src/main.js [hunk 2]
```diff
@@ -255,6 +262,27 @@ table+table[aria-hidden=true] {
...
+            var dateValue = new Date(parts[0], --parts[1], parts[2]);
+
+            if (!isNaN(dateValue.getTime())) {
+                return dateValue;
+            }
+
+            return null;
+        },
+        _setValueAsDate(dateValue) {
+            if (isNaN(dateValue.getTime())) {
```
[the remark this answers is not in the document]
Comment (abaranovskii) [reply src/main.js]: @chemerisuk yep, fixed.

**Comment by chemerisuk** [comment]
👍

**Pull request merged**